tests/test_application.coffee

```diff
@@ -87,20 +106,42 @@ Space.Error.extend CustomerApp, 'InvalidCustomerName', {
 
 class CustomerApp.Customer extends Space.eventSourcing.Aggregate
 
+  dependencies: {
+    myCommandDependency: 'myCommandDependency'
+    myEventDependency: 'myEventDependency'
+  }
+
   fields: {
     name: String
   }
 
   commandMap: -> {
     'CustomerApp.CreateCustomer': (command) ->
+      @myCommandDependency()
+
       @record new CustomerApp.CustomerCreated {
         sourceId: @getId()
         customerName: command.name
       }
+
+    'CustomerApp.ChangeCustomerName': (command) ->
+      @myCommandDependency()
+
+      @record new CustomerApp.CustomerNameChanged {
+        sourceId: @getId()
+        customerName: command.name
+      }
   }
 
   eventMap: -> {
-    'CustomerApp.CustomerCreated': (event) -> @name = event.customerName
+    'CustomerApp.CustomerCreated': (event) ->
+      @myEventDependency()
```

Comment:
This is looking good

> events on both paths still have access to dependencies.

This is expected, it's up to the implementor to know where *not* to use deps :-). We will need to have clear instructions for:
• aggregate state changing event handlers should *never* use an external dependency
• provide use-cases for _when_ to use an external dependency prior to state-change

At a later time it would be ideal to [abstract the internal state-change event mapping](https://github.com/meteor-space/event-sourcing/issues/60), as we could then just provide a mapping without the option of accessing a dependency.